### PR TITLE
migrate pkg/scheduler/framework/plugins/volume to structured logs

### PR DESCRIPTION
**Pull request by chymy**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
xref:
https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/scheduler/framework/plugins/volumezone/volume_zone.go [hunk 1]
```diff
@@ -158,7 +158,7 @@ func (pl *VolumeZone) Filter(ctx context.Context, _ *framework.CycleState, pod *
 			nodeV, _ := nodeConstraints[k]
 			volumeVSet, err := volumehelpers.LabelZonesToSet(v)
 			if err != nil {
-				klog.Warningf("Failed to parse label for %q: %q. Ignoring the label. err=%v. ", k, v, err)
+				klog.ErrorS(err,"Failed to parse label, ignoring the label", k, v)
```
Comment (gavinfish) [inline pkg/scheduler/framework/plugins/volumezone/volume_zone.go:161]: I think info here is enough. And `"label", fmt.Printf("%q: %q",k,v)` should be more reasonable.

**Review by gavinfish — COMMENTED** [review]
/ok-to-test

File: pkg/scheduler/framework/plugins/volumezone/volume_zone.go (hunk 1 above)
Comment (chymy) [reply pkg/scheduler/framework/plugins/volumezone/volume_zone.go]: Thanks @gavinfish , it's `fmt.Sprintf`. fixed.

File: pkg/scheduler/framework/plugins/volumebinding/volume_binding.go [hunk 2]
```diff
@@ -254,13 +254,13 @@ func (pl *VolumeBinding) PreBind(ctx context.Context, cs *framework.CycleState,
 	if !ok {
 		return framework.AsStatus(fmt.Errorf("no pod volumes found for node %q", nodeName))
 	}
-	klog.V(5).Infof("Trying to bind volumes for pod \"%v/%v\"", pod.Namespace, pod.Name)
+	klog.V(5).InfoS("Trying to bind volumes for pod", "pod", klog.KRef(pod.Namespace, pod.Name))
```
Comment (alculquicondor) [inline pkg/scheduler/framework/plugins/volumebinding/volume_binding.go:257]: Use `klog.KObj` instead

File: pkg/scheduler/framework/plugins/volumebinding/volume_binding.go [hunk 3]
```diff
@@ -254,13 +254,13 @@ func (pl *VolumeBinding) PreBind(ctx context.Context, cs *framework.CycleState,
...
 	err = pl.Binder.BindPodVolumes(pod, podVolumes)
 	if err != nil {
-		klog.V(1).Infof("Failed to bind volumes for pod \"%v/%v\": %v", pod.Namespace, pod.Name, err)
+		klog.V(1).InfoS("Failed to bind volumes for pod", "pod", klog.KRef(pod.Namespace, pod.Name), "err", err)
```
Comment (alculquicondor) [inline pkg/scheduler/framework/plugins/volumebinding/volume_binding.go:260]: @cofyc why was this just InfoS?

Also, errors that are returned are already logged up the stack

File: pkg/scheduler/framework/plugins/volumebinding/volume_binding.go (hunk 2 above)
Comment (chymy) [reply pkg/scheduler/framework/plugins/volumebinding/volume_binding.go]: Thanks, done.

File: pkg/scheduler/framework/plugins/volumebinding/volume_binding.go [hunk 4]
```diff
@@ -254,13 +254,13 @@ func (pl *VolumeBinding) PreBind(ctx context.Context, cs *framework.CycleState,
 	if !ok {
 		return framework.AsStatus(fmt.Errorf("no pod volumes found for node %q", nodeName))
 	}
-	klog.V(5).Infof("Trying to bind volumes for pod \"%v/%v\"", pod.Namespace, pod.Name)
+	klog.V(5).InfoS("Trying to bind volumes for pod", "pod", klog.KObj(pod))
 	err = pl.Binder.BindPodVolumes(pod, podVolumes)
 	if err != nil {
-		klog.V(1).Infof("Failed to bind volumes for pod \"%v/%v\": %v", pod.Namespace, pod.Name, err)
+		klog.V(1).ErrorS(err, "Failed to bind volumes for pod", "pod", klog.KObj(pod))
```
Comment (alculquicondor) [inline pkg/scheduler/framework/plugins/volumebinding/volume_binding.go:260]: Leave this as `InfoS`, put the error as argument with `err` key

File: pkg/scheduler/framework/plugins/volumebinding/volume_binding.go [hunk 5]
```diff
@@ -254,13 +254,13 @@ func (pl *VolumeBinding) PreBind(ctx context.Context, cs *framework.CycleState,
...
 		return framework.AsStatus(err)
 	}
-	klog.V(5).Infof("Success binding volumes for pod \"%v/%v\"", pod.Namespace, pod.Name)
+	klog.V(5).InfoS("Success binding volumes for pod", "pod", klog.KRef(pod.Namespace, pod.Name))
```
Comment (alculquicondor) [inline pkg/scheduler/framework/plugins/volumebinding/volume_binding.go:263]: use `klog.KObj`

File: pkg/scheduler/framework/plugins/volumezone/volume_zone.go [hunk 6]
```diff
@@ -158,7 +158,7 @@ func (pl *VolumeZone) Filter(ctx context.Context, _ *framework.CycleState, pod *
 			nodeV, _ := nodeConstraints[k]
 			volumeVSet, err := volumehelpers.LabelZonesToSet(v)
 			if err != nil {
-				klog.Warningf("Failed to parse label for %q: %q. Ignoring the label. err=%v. ", k, v, err)
+				klog.ErrorS(err, "Failed to parse label, ignoring the label", "label", fmt.Sprintf("%q: %q", k, v))
```
Comment (alculquicondor) [inline pkg/scheduler/framework/plugins/volumezone/volume_zone.go:161]: `InfoS` as well

File: pkg/scheduler/framework/plugins/volumezone/volume_zone.go [hunk 7]
```diff
@@ -158,7 +158,7 @@ func (pl *VolumeZone) Filter(ctx context.Context, _ *framework.CycleState, pod *
 			nodeV, _ := nodeConstraints[k]
 			volumeVSet, err := volumehelpers.LabelZonesToSet(v)
 			if err != nil {
-				klog.Warningf("Failed to parse label for %q: %q. Ignoring the label. err=%v. ", k, v, err)
+				klog.InfoS("Failed to parse label, ignoring the label", "label", fmt.Sprintf("%q: %q", k, v), "err", err)
```
[the remark this answers is not in the document]
Comment (chymy) [reply pkg/scheduler/framework/plugins/volumezone/volume_zone.go]: Thanks @alculquicondor , done.

[the remark this answers is not in the document]
Comment (alculquicondor) [reply pkg/scheduler/framework/plugins/volumezone/volume_zone.go]: remove the space

[the remark this answers is not in the document]
Comment (chymy) [reply pkg/scheduler/framework/plugins/volumezone/volume_zone.go]: ok, fixed. Thanks.

**Review by alculquicondor — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**